server/src/main/java/org/apache/cassandra/sidecar/lifecycle/ProcessLifecycleProvider.java

```diff
@@ -18,38 +18,313 @@
+
+            Long pid = readPidFromFile(pidFilePath);
+            Optional<ProcessHandle> processHandle = ProcessHandle.of(pid);
+
+            // Case 2: No process running with such PID or process is not alive
+            if (processHandle.isEmpty() || !processHandle.get().isAlive())
+            {
+                LOG.debug("No running process found with PID {} for instance {}", pid, instance.host());
+                deletePidFile(instance, pidFilePath);
+                return false;
+            }
+            
+            // Case 3: Process with such PID is running - check if it's a Cassandra process
+            // If we can't determine the command line, we assume it's Cassandra
+            Optional<String> cmdLine = getCommandLinePlatformIndependent(processHandle.get());
+            if (cmdLine.isEmpty() || cmdLine.get().contains("org.apache.cassandra.service.CassandraDaemon"))
+            {
+                LOG.debug("Cassandra instance {} is running with PID {}", instance.host(), pid);
+                return true;
+            }
+            
+            // Case 4: Process with such PID is running but it's not a Cassandra process
+            LOG.debug("Process with PID {} for instance {} is not a Cassandra process (command line: {}).",
+                        pid, instance.host(), cmdLine);
+            deletePidFile(instance, pidFilePath);
+            return false;
+        }
+        catch (Exception e)
+        {
+            LOG.warn("Failed to read PID from file {} for instance {}: {}", pidFilePath, instance.host(), e.getMessage());
+            return false;
+        }
+    }
+
+    protected static void deletePidFile(InstanceMetadata instance, Path pidFilePath)
+    {
+        try
+        {
+            LOG.info("Deleting stale PID file {} for instance {}", pidFilePath, instance.host());
+            Files.delete(pidFilePath);
+        } 
+        catch (Exception e)
+        {
+            LOG.warn("Failed to delete stale PID file {} for instance {}: {}", pidFilePath, instance.host(), e.getMessage());
+        }
+    }
+
+    /*
+     * Due to JDK-8345117 java can sometimes truncate the command line returned by ProcessHandle.info().commandLine()
+     * on some platforms (ie. Linux). To work around this, we use the 'ps' command to get the full command line.
+     * This method should be platform-independent as it relies on the 'ps' command which is available on most Unix-like systems.
+     * For non-Unix systems, we fall back to the default implementation.
+    */
+    protected static Optional<String> getCommandLinePlatformIndependent(ProcessHandle processHandle)
+    {
+        long pid = processHandle.pid();
+        try
+        {
+            ProcessBuilder pb = new ProcessBuilder("ps", "-p", String.valueOf(pid), "-o", "args=");
+            Process proc = pb.start();
+            try (BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(proc.getInputStream())))
```

Comment:
To address the spotbugs error
```suggestion
                    new InputStreamReader(proc.getInputStream(), StandardCharsets.UTF_8)))
```

Reply:
updated on [b643899](https://github.com/apache/cassandra-sidecar/pull/277/commits/b643899ce6968dc05b0cb9fcb7ea51c1bd54021a)